tests/foreman/cli/test_product.py

```diff
@@ -246,3 +247,101 @@ def test_product_list_with_default_settings(self):
             Defaults.delete({u'param-name': 'organization_id'})
             result = ssh.command('hammer defaults list')
             self.assertTrue(default_org['id'] not in "".join(result.stdout))
+
+    @tier2
+    def test_positive_assign_http_proxy_to_products(self):
+        """Assign http_proxy to Products and check whether http-proxy is
+         used during sync.
+
+        :id: 6af7b2b8-15d5-4d9f-9f87-e76b404a966f
+
+        :expectedresults: HTTP Proxy is assigned to all repos present
+            in Products and sync operation uses assigned http-proxy.
+
+        :CaseImportance: Critical
+        """
+        # create HTTP proxies
+        http_proxy_url_a = '{}:{}'.format(
+            gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999))
+        http_proxy_url_b = '{}:{}'.format(
+            gen_url(scheme='https'), gen_integer(min_value=10, max_value=9999))
+        http_proxy_a = HttpProxy.create({
+            'name': gen_string('alpha', 15),
+            'url': http_proxy_url_a,
+            'organization-id': self.org['id'],
+        })
+        http_proxy_b = HttpProxy.create({
+            'name': gen_string('alpha', 15),
+            'url': http_proxy_url_b,
+            'organization-id': self.org['id'],
+        })
+        proxy_fqdn = http_proxy_b['url'].split(":")[1].strip("//")
+        # Create products and repositories
+        product_a = make_product({
+            u'organization-id': self.org['id'],
+        })
+        product_b = make_product({
+            u'organization-id': self.org['id'],
+        })
+        repo_a1 = make_repository({
+            'product-id': product_a['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'none'
+        })
+        repo_a2 = make_repository({
+            'product-id': product_a['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'use_selected_http_proxy',
+            'http-proxy-id': http_proxy_a['id']
+        })
+        repo_b1 = make_repository({
+            'product-id': product_b['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'none'
+        })
+        repo_b2 = make_repository({
+            'product-id': product_b['id'],
+            'url': FAKE_0_YUM_REPO,
+            'http-proxy-policy': 'global_default_http_proxy',
```

Comment:
By default behaviour is `global_default_http_proxy` when a repo creates .. hence we do not need to specify that here ..

Reply:
Updated PR.